pennylane/ops/functions/dot.py

```diff
@@ -104,6 +104,32 @@ def dot(
         <lambda>(params_0, t) * X(0)
       + <lambda>(params_1, t) * Y(0)
     )
+
+    .. details::
+        :title: Grouping
+
+        Grouping information can be collected during construction using the ``grouping_type`` and ``method``
+        keyword arguments. For example:
+
+        .. code-block:: python
+
+            import pennylane as qml
+
+            a = qml.PauliX(0)
+            b = qml.PauliX(1)
```

Comment:
```suggestion
            b = qml.PauliX(0) @ qml.PauliX(1)
```

feel free to ignore, but I think this will make it a bit more clear since before the grouping is ambiguous with also ((0,), (1, 2)) being a valid grouping